hal/src/flash.rs

```diff
@@ -403,6 +406,81 @@ impl<'a> Flash<'a> {
         ret
     }
 
+    /// Program a user-defined type.
+    ///
+    /// # Safety
+    ///
+    /// 1. Do not write to flash memory that is being used for your code.
+    /// 2. The destination address must be within the flash memory region.
+    /// 3. The `from` and `to` pointers must be aligned to the pointee type.
```

Comment:
I took the safety command of the standard_program function, since I didn't really know what the safety concerns would be for this function. This make a more sense.